### PR TITLE
[processor/transform] Update traces to use tqltraces

### PR DESCRIPTION
**Pull request by TylerHelmuth**
**Description:** 
Updates traces to use the tqltraces context from `pkg/telemetryquerylanguage`.  No functional change to the transform processor.

**Link to tracking Issue:** 
Related to #11751 

**Testing:** 
Ran unit tests

### CODE REVIEW

**Review by bogdandrutu — APPROVED** [review]

File: cmd/configschema/go.mod [hunk 1]
```diff
@@ -544,6 +544,7 @@ require (
 	go.opentelemetry.io/otel/metric v0.31.0 // indirect
 	go.opentelemetry.io/otel/sdk v1.8.0 // indirect
 	go.opentelemetry.io/otel/trace v1.8.0 // indirect
+	go.opentelemetry.io/proto/otlp v0.18.0 // indirect
```
Comment (bogdandrutu) [inline cmd/configschema/go.mod:547]: Where is this coming from?

Comment (TylerHelmuth) [reply cmd/configschema/go.mod:547]: I am not sure yet, I am surprised to see that as well.  If I run `make gotidy` it shows up.

Comment (TylerHelmuth) [reply cmd/configschema/go.mod:547]: I'll look into it more tomorrow

Comment (TylerHelmuth) [reply cmd/configschema/go.mod:547]: @bogdandrutu it is coming from the fact that we use the proto libraries to help build our enum symbol table: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/d00fbea7b96e75d67571696deccb2909cef806b6/pkg/telemetryquerylanguage/contexts/tqltraces/traces.go#L64

Comment (bogdandrutu) [reply cmd/configschema/go.mod:547]: You should use `ptrace.SpanKind[Unspecified|Internal|...]`

Comment (bogdandrutu) [reply cmd/configschema/go.mod:547]: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/12905

**Review by codeboten — COMMENTED** [review]
Please rebase

**Pull request merged**